File: W14/progLag2/Set.hs
data Set a = Empty | Set [a]
	deriving (Show,Eq)

member e Empty = False
member e (Set xs) = elem e xs

insert e (Set xs)
	| member e (Set xs) == True = Set xs
	| otherwise = Set (e:xs)

remove e Empty = Empty
remove e (Set xs) = Set (filter (==e) xs) 

File: W14/progLag2/Set.java
import java.util.*;

public class Set {
  private ArrayList<Object> s;
  
  Set() {
    this.s = new ArrayList<Object>();
  }
  
  public boolean member(Object e) {
    for (int i=0; i<this.s.size(); i++) {
         if (this.s.get(i) == e) {
           return true;
         }
     }
    return false;
  }
  
  public void insert(Object e) {
    for (int i=0; i<this.s.size(); i++) {
		if (this.s.get(i) == e) {
			this.s.add(e);
    	}
  	}
  }
  
  public void remove(Object e) {
    for (int i=0; i<this.s.size(); i++) {
         if (this.s.get(i) == e) {
           this.s.remove(i);
         }
    }
  }
}
      
         
